geyser-plugin-manager/src/transaction_notifier.rs

```diff
@@ -90,12 +90,122 @@ impl TransactionNotifierImpl {
         transaction_status_meta: &'a TransactionStatusMeta,
         transaction: &'a SanitizedTransaction,
     ) -> ReplicaTransactionInfoV2<'a> {
+        let msg = transaction.message();
+        let instructions = msg.instructions();
+        let account_keys = msg.account_keys();
+
+        // Detect the two-instruction durable-nonce advance + vote pattern
+        let durable_nonce_and_vote = instructions.len() == 2
+            && account_keys[instructions[0].program_id_index as usize]
+                == solana_system_program::id()
+            && account_keys[instructions[1].program_id_index as usize] == solana_vote_program::id();
+
```

Comment:
could you remind me which validators send votes in this format? 🤔 also, is_vote is a bit subtle. if someone just inserts a vote instruction into a random transaction, does it still count as a vote transaction?

cc @t-nelson @lijunwangs